src/OpenTK/Platform/X11/XI2MouseKeyboard.cs

```diff
@@ -397,6 +399,14 @@ private void UpdateDevices()
                     }
                     XI.FreeDeviceInfo((IntPtr)list);
                 }
+
+                // Resture current mouse position
+                if (devices.Count >= 1) {
+                    devices[0].State.Position = current_mouse_state.Position;
+                    devices[0].State.SetScrollAbsolute(
+                            current_mouse_state.Scroll.X,
+                            current_mouse_state.Scroll.Y);
+                }
```

Comment:
Why do we need to reset the position here?
Is the 0th device special or what is it that is going on?
Are we applying events from other devices to the 0th device?

Reply:
> Why do we need to reset the position here?

This function, UpdateDevices(), is called whenever a DeviceChanged event is received. This function removes and re-adds all devices and thus resets the current mouse position to the default (0, 0). The intent of this is to save and restore the current mouse position instead of resetting to (0, 0).

> Is the 0th device special or what is it that is going on?

The 0th device is not really special here except that I needed to save the position somewhere and it was the most convenient place. My thinking regarding this:

The IMouseDriver2.GetState() function merges all of the state information together, so putting them into a single device is more-or-less equivalent.

I'm not able to find a system that actually supports multiple mouse devices. As far as I can tell, both Xorg and XWayland emulate a single mouse even when multiple mice are present so I don't think there's any regression caused by doing it this way.

Reply:
Instead of this "Save mouse state" patch, I do have an alternate approach I'm working on that you may find cleaner. Instead of calling UpdateDevices() on a DeviceChanged event and saving/restoring the position, it processes the DeviceChanged event and updates the configuration of that device.

Reply:
From what I understand from your explanation I think the keeping of the position and scroll info is a quite efficient way to solve the problem.
I like that the risk of this change affecting other parts of the software are quite low.

From the sound of your cleaner solution it might require more complicated logic that is harder to prove.
But I might be wrong in that assessment, and I'm also quite fond of "correct" solutions so I think it could be worth a look if it's not a ton of work on your part.

I just want to say I appreciate you taking the time to make this PR. It's helps a lot.